wp-content/themes/twentyseventeen-wp20/style.css

```diff
@@ -696,132 +699,94 @@ body:not(.title-tagline-hidden) .site-branding-text {
  * Locale Switcher
  */
 
-.navigation-top-menu-container {
+.wp20-locale-switcher-container {
+	flex: 1;
 	position: relative;
+	text-align: end;
 }
 
-.navigation-top-menu-container > * {
-	position: relative !important;
-	width: 100% !important;
-	left: 0 !important;
-	top: 0 !important;
-}
-
-.navigation-top-menu-container .select2-dropdown {
-	border-top: 1px solid var(--wp20--color--border) !important;
-}
-
-@media screen and (min-width: 769px) {
-	.navigation-top-menu-container > * {
-		position: absolute !important;
-		left: auto !important;
-		right: 0 !important;
-		width: 50% !important;
-		max-width: 250px;
-	}
-
-	.navigation-top-menu-container .select2-dropdown {
-		border: 1px solid var(--wp20--color--border) !important;
-	}
-}
-
-.wp20-locale-switcher-container {
+.wp20-locale-switcher-container form {
 	position: absolute;
```

Comment:
This makes it so that `.site-content-contain` is covered by the dropdown on mobile:

<img width="362" alt="Screenshot 2023-05-03 at 7 44 56 AM" src="https://user-images.githubusercontent.com/484068/235951491-0115203e-fcbe-4c49-b444-d9b64ecefe63.png">

On `trunk` it pushes the content down though:

<img width="409" alt="Screenshot 2023-05-03 at 7 47 46 AM" src="https://user-images.githubusercontent.com/484068/235952231-67ecfae2-73f4-43f9-94d6-b2a1900726a5.png">

I don't see a good way to fix that. One problem I kept running into with these PRs is that the menu DOM structure is different than the locale list structure. The menu toggle button is a separate element from the menu itself, while the switcher "toggle" and dropdown are all contained in a single element.

Another part might be that [Accessible Autocomplete doesn't add classes for the current state](https://github.com/alphagov/accessible-autocomplete/issues/384). The `input` can be styles by matching `aria-expanded=true`, though; and the `ul` with `.autocomplete__menu--visible`.

Reply:
I think this is fixed now. I've added a wrapper to the nav menu so that it behaves more like the locale one. Try it out.